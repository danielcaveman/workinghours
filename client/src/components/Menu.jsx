import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";
import logo from "../assets/logo.png";

const MenuContainer = styled.aside`
  grid-area: menu;
  background-color: #252d32;
  padding: 2rem 1rem;
`;

const Logo = styled.div`
  background-image: url(${logo});
  background-size: cover;
  height: 3rem;
  width: 20rem;
  margin: 0 auto;
`;

const Nav = styled.nav`
  padding: 3rem 0;
  margin: 2rem 1rem;
  border-top: 1px solid #6e6a6a;
`;

const List = styled.ul`
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  list-style: none;
  font-size: 1.7rem;
`;

const ListItem = styled.li``;

const Link = styled(NavLink)`
  padding: 1rem 2.5rem;
  color: #fff;
  cursor: pointer;
  display: block;
  border-left: 0.4rem solid #5eabf2;
  text-decoration: none;

  &:hover {
    background-color: #5eabf2;
  }
`;

function Menu() {
  return (
    <MenuContainer>
      <Logo />
      <Nav>
        <List>
          <ListItem>
            <Link to="/">Main Page</Link>
          </ListItem>
        </List>
      </Nav>
    </MenuContainer>
  );
}

export default Menu;
