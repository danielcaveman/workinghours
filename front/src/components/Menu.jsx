import styled from "styled-components";
import logo from "../assets/logo.png";

const MenuContainer = styled.aside`
  grid-area: menu;
  background-color: #252d32;
  padding: 2rem 1rem;
`;

const Logo = styled.div`
  background-image: url(${logo});
  background-size: cover;
  height: 3rem;
  width: 20rem;
  margin: 0 auto;
`;

function Menu() {
  return (
    <MenuContainer>
      <Logo />
    </MenuContainer>
  );
}

export default Menu;
