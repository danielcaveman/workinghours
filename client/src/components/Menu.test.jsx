import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Menu from "./Menu";

describe("<Menu />", () => {
  it("Should render menu component correctly", () => {
    render(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
  });

  it("Should match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
