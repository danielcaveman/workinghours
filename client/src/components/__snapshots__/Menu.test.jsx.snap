// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should match snapshot 1`] = `
<div>
  <aside
    class="sc-bdfBwQ kvXtJj"
  >
    <div
      class="sc-gsTCUz LQcIy"
    />
    <nav
      class="sc-dlfnbm efGIcZ"
    >
      <ul
        class="sc-hKgILt dCLLof"
      >
        <li
          class="sc-eCssSg"
        >
          <a
            aria-current="page"
            class="sc-jSgupP hgGPFK active"
            href="/"
          >
            Main Page
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
